cpp/src/arrow/array/array_primitive.h

```diff
@@ -119,8 +129,6 @@ class NumericArray : public PrimitiveArray {
   IteratorType end() const { return IteratorType(*this, length()); }
 
  protected:
-  using PrimitiveArray::PrimitiveArray;
```

Comment:
Do we need to add `NumericArray()` if this line is deleted?

Reply:
Nice catch 🤔